cardano_node_tests/tests/test_plutus_v2_spend_build.py

```diff
@@ -1462,3 +1498,464 @@ def test_v1_attached_v2_reference(
             )
         err_str = str(excinfo.value)
         assert "ReferenceInputsNotSupported" in err_str, err_str
+
+
+@pytest.mark.testnets
+class TestReferenceInputs:
```

Comment:
Make it clear in test class / test function names and docstrings that these tests are for **readonly** reference txins (i.e. txins that will be added to script context), not for reference txins that are actually used for spending.